src/components/Dropdown/index.tsx

```diff
@@ -1,166 +1,178 @@
-  const detailsRef = useRef<HTMLDetailsElement | null>(null);
-
-  const closeDropdown = (e: MouseEvent) => {
-    const { target } = e;
-    if (!target || !detailsRef.current) return;
-    detailsRef.current.open = false;
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+  const [triggerSize, setTriggerSize] = useState<{
+    width: number;
+    height: number;
+  }>({ width: 0, height: 0 });
+
+  const contextValues = {
+    isOpen,
+    setIsOpen,
+    collapseOnBlur,
+    dropdownLabel,
+    id,
+    direction,
+    triggerSize,
+    setTriggerSize,
   };
 
   return (
-    <DropdownWrapper ref={detailsRef}>
-      <summary>{toggleElement}</summary>
-      <DropdownList onClick={closeDropdown} align={align}>
+    <DropdownContext.Provider value={contextValues}>
+      <Container
+        css={css`
+          width: auto;
+          height: auto;
+          position: relative;
+          overflow: visible;
+        `}
+      >
         {children}
-      </DropdownList>
-    </DropdownWrapper>
+      </Container>
+    </DropdownContext.Provider>
   );
 };
 
-interface DropdownItemProps extends DefaultProps<HTMLLIElement> {
-  title?: string;
-  leftIcon?: IconType;
-  rightIcon?: IconType;
-  description?: string;
-  onClick?: MouseEventHandler;
-}
+const Trigger = ({ children }: DefaultProps<HTMLDivElement>) => {
+  const { isOpen, setIsOpen, dropdownLabel, id, setTriggerSize } =
+    useContext(DropdownContext);
+  const triggerRef = useRef<HTMLDivElement | null>(null);
+
+  useEffect(() => {
+    if (!triggerRef.current || !setTriggerSize) return;
+    setTriggerSize({
+      width: triggerRef.current.offsetWidth,
+      height: triggerRef.current.offsetHeight,
+    });
+  }, [triggerRef.current]);
 
-const DropdownItem = ({
-  title = '',
-  leftIcon: LeftIcon,
-  rightIcon: RightIcon,
-  description = '',
-  onClick: clickEventHandler,
-}: DropdownItemProps) => {
   return (
-    <DropdownListItem onClick={clickEventHandler}>
-      {LeftIcon && (
-        <Flexbox alignItems="center">
-          <LeftIcon size={'1.2rem'} />
-        </Flexbox>
+    <>
+      {setIsOpen && (
+        <div
+          onClick={(e: ReactMouseEvent) => {
+            e.stopPropagation();
+            setIsOpen(!isOpen);
+          }}
+          aria-expanded={isOpen}
+          aria-haspopup="true"
+          aria-label={dropdownLabel}
+          id={id}
+          ref={triggerRef}
+        >
+          {children}
+        </div>
       )}
-      <Flexbox
-        flexDirection="column"
-        gap=".3rem"
-        alignItems="center"
-        css={css`
-          flex: 1;
-          text-align: left;
-        `}
-      >
-        {title && <ItemTitle>{title}</ItemTitle>}
-        {description && <ItemDescription>{description}</ItemDescription>}
-      </Flexbox>
-      {RightIcon && (
-        <Flexbox alignItems="center">
-          <RightIcon size={'1.2rem'} />
-        </Flexbox>
-      )}
-    </DropdownListItem>
+    </>
   );
```

Comment:
```suggestion
  const onClick = (e: ReactMouseEvent) => {
    e.stopPropagation();

    if (!setIsOpen) return;
    setIsOpen(!isOpen);
  };

  return (
    <div
      onClick={onClick}
      aria-expanded={isOpen}
      aria-haspopup="true"
      aria-label={dropdownLabel}
      id={id}
      ref={triggerRef}
    >
      {children}
    </div>
  );
```

단순 타입가드면 setIsOpen 컴포넌트 렌더링의 조건이 되는것보다 이렇게 쓰는게 좋지 않을까요?

Reply:
앗 이게 사실 사연이 있는데.. `setIsOpen`을 포함한 context 값들이 null이면 저 div를 렌더링할 수가 없어서 저런식으로 nullish 체크를 했어요.. 더 이쁜 방법이 필요해요